rllib/utils/actor_manager.py

```diff
@@ -0,0 +1,324 @@
+from collections import defaultdict
+import logging
+from typing import Callable, List, Optional, Tuple, TypeVar
+
+import ray
+from ray.actor import ActorHandle
+from ray.exceptions import RayActorError
+from ray.util.annotations import DeveloperAPI
+
+
+logger = logging.getLogger(__name__)
+# Generic type var for foreach_* methods.
+T = TypeVar("T")
+
+
+class FaultTolerantActorManager:
+    """A manager that is aware of the healthiness states of remote actors.
+
+    Args:
+        actors: A list of ray remote actors to manage on. These actors must have an
+            ``apply`` method which takes a function with no parameters.
+        max_remote_requests_in_flight_per_actor: The maximum number of remote
+            requests that can be in flight per actor. Any requests made to the pool
+            that cannot be scheduled because the limit has been reached will be dropped.
+            This only applies to the async mode.
+    """
+
+    def __init__(
+        self,
+        actors: Optional[List[ActorHandle]] = None,
+        max_remote_requests_in_flight_per_actor: Optional[int] = 2,
+        ignore_worker_failure=True,
+    ):
+        self.__actors = actors or []
+        # Start with active state for all remote actors.
+        self.__remote_actor_states = [True] * len(self.__actors)
+
+        # Collection of outstanding async requests.
+        self.__remote_reqs_in_flight = defaultdict(lambda: 0)
+        self.__in_flight_req_to_actor_idx = {}
```

Comment:
Type is a `Mapping[ObjRef, int]` please clarify this here. I had to read the code to understand this.